js/core.js

```diff
@@ -1046,9 +1047,39 @@ function parsePreamble(preamble, key, value, lineno) {
     }
 }
 
-function parsePhraseBook(s) {
+function parseImport(preamble, sketchKey, refName, importcode){
+    preamble["sketchKey"] = sketchKey;
+    return parsePhraseBook(importcode, false);
+    // let s = "root:\n- Kunal boi";
+    // return parsePhraseBook(s);
+}
+
+function checkImport(code){
+    const lines = code.split('\n');
+
+    for (let i = 0; i < lines.length; i++) {
+        const line = lines[i];
+        const trimmedLine = line.trim();
+
+        if (trimmedLine[0] === '#') {
+            let postHash = line.slice(1).match(IMPORT_RE);
+            if(postHash){
+                return postHash[2];
+            }
+            // else if(trimmedLine.length !== 0){
+            //     throw new Error(`Line ${ i+1 }: expecting '#import "sketch key" as reference-name'`);
+            // }
+            else return false;
+        }
+    }
+}
+
+function parsePhraseBook(s,importcode) {
```

Comment:
Ok, if I understand correctly, you are pre-parsing the code looking for an import statement, and then load the sketch and the feed this to the parser. I don't think this is the way to go. There could be more than one 'import' statement. And an imported sketch could itself import another sketch.